cli/disk_cache.rs

```diff
@@ -50,7 +50,7 @@ impl DiskCache {
     out.push(scheme);
 
     match scheme {
-      "http" | "https" => {
+      "http" | "https" | "wasm" => {
```

Comment:
This function is in `disk_cache` - there are other places that verify URL scheme and they definitely do not take `wasm://` scheme. (I didn't know that `wasm://` scheme is a thing)

Reply:
@caspervonb do you know how problematic URL looks like and where does it originate from? Problem you described in issue looks strange; as process should crash before hitting `disk_cache` if there was `wasm://` scheme.

Reply:
> This function is in disk_cache - there are other places in SourceFileFetcher and ModuleGraph that verify URL scheme and they definitely do not take wasm:// scheme.

@bartlomieju seems to re-enter Deno at `deno_core::isolate::exception_to_err_result`.

> Problem you described in issue looks strange; as process should crash before hitting disk_cache if there was wasm:// scheme.

It does panic; trace was in the issue but here it is again

```
  0:        0x10adfe1ec - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1216a26f3ce3c114
   1:        0x10ae1cbae - core::fmt::write::ha0ec7ad9e1aa7104
   2:        0x10adfd929 - std::io::Write::write_fmt::he79f4e513ba146b5
   3:        0x10adef49a - std::panicking::default_hook::{{closure}}::hdf6a192f990c9642
   4:        0x10adef1a5 - std::panicking::default_hook::hc1d0cb86aebc9fbd
   5:        0x10adefb08 - std::panicking::rust_panic_with_hook::hbe9b3056eff949a6
   6:        0x10adef6d2 - rust_begin_unwind
   7:        0x10ae4298b - std::panicking::begin_panic_fmt::hb770b3908b606330
   8:        0x109bc117e - deno::disk_cache::DiskCache::get_cache_filename_with_extension::h9bead961694c7bab
   9:        0x109b0080f - deno::tsc::TsCompiler::get_source_map_file::he10d1f955e964df3
  10:        0x109b00db8 - <deno::tsc::TsCompiler as deno::source_maps::SourceMapGetter>::get_source_map::h9a57a5b6593e518a
  11:        0x1099f068b - deno::source_maps::get_orig_position::h0fcac86f7ca8ea7d
  12:        0x1099efe1c - deno::source_maps::apply_source_map::ha23dbe5088bcb2f3
  13:        0x109b3520c - deno::worker::Worker::new::{{closure}}::h1031ac6f522a4c0e
  14:        0x10a237a47 - deno_core::isolate::exception_to_err_result::he185b15303110714
  15:        0x10a2303bc - deno_core::es_isolate::EsIsolate::mod_evaluate::ha4e54ab131b37f04
  16:        0x109a9df52 - <std::future::GenFuture<T> as core::future::future::Future>::poll::hef4c9d64af759bf3
  17:        0x109a72fbe - <std::future::GenFuture<T> as core::future::future::Future>::poll::h13505288ce1a253d
  18:        0x109a2d958 - std::thread::local::LocalKey<T>::with::hdd75ccf8bc47ed44
  19:        0x109bba4fe - tokio::macros::scoped_tls::ScopedKey<T>::set::h9b3141d2c27c8389
  20:        0x109c50af1 - tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::ha0be0ef39a82a650
  21:        0x109bbd332 - tokio::runtime::context::enter::h0ff9f498deb68362
  22:        0x109a1ea46 - deno::tokio_util::run_basic::h3538e974f25d293c
  23:        0x109b04a11 - deno::main::h5f6baf7a364ebade
  24:        0x109c1ee46 - std::rt::lang_start::{{closure}}::h9ec716e45ba52b55
  25:        0x10adef5d8 - std::panicking::try::do_call::h69374c200cb9b71d
  26:        0x10ae060cb - __rust_maybe_catch_panic
  27:        0x10adea733 - std::rt::lang_start_internal::h892855ab5abc97d0
  28:        0x109b08509 - main```

Reply:
@caspervonb thanks, can you run with `-Ldebug` flag and dump output? I'm curious to see what's the problematic URL. 

It seems that requesting source map with URL with `wasm://` scheme is a special case because of mechanism you described above:

> These are traps; which are basically panics that WebAssembly cannot handle and delegate to the host which then in turn represents them as exceptions.

> But yes; in Chrome these stack traces are clickable links in the console that can be opened in the debugger which presents it in the s-expression form; debug builds with DWARF get the full source mapping experience.

Reply:
```
$ ./target/debug/deno run -Ldebug cli/tests/wasm_unreachable.ts 
Deno isolate init with snapshots.
rust:shared_queue:reset
DEBUG JS - cwd /Users/caspervonb/Projects/deno
DEBUG JS - args []
main_module file:///Users/caspervonb/Projects/deno/cli/tests/wasm_unreachable.ts
fetch_source_file specifier: file:///Users/caspervonb/Projects/deno/cli/tests/wasm_unreachable.ts 
⚠️️  Granted read access to "/Users/caspervonb/Projects/deno/cli/tests/wasm_unreachable.ts"
compiled filename: "/Users/caspervonb/Library/Caches/deno/gen/file/Users/caspervonb/Projects/deno/cli/tests/wasm_unreachable.ts.js"
register_complete file:///Users/caspervonb/Projects/deno/cli/tests/wasm_unreachable.ts
DEBUG JS - sendSync op_apply_source_map 5
source map filename: "/Users/caspervonb/Library/Caches/deno/gen/file/Users/caspervonb/Projects/deno/cli/tests/wasm_unreachable.ts.js.map"
thread 'main' panicked at 'not implemented: Don't know how to create cache name for scheme: wasm', cli/disk_cache.rs:97:9
```